File: src/main/java/com/jayden/monitoring/user/UserService.java
package com.jayden.monitoring.user;

import com.jayden.monitoring.db.entity.User;
import com.jayden.monitoring.db.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    @Transactional
    public User createUser(CreateUserRequest request) {
        User user = request.toEntity();
        userRepository.save(user);
        return user;
    }

    public List<User> getUserList() {
        return userRepository.findAll();
    }

    public User getUser(Long id) {
        return userRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Not Found User"));
    }
}

File: src/main/kotlin/com/jayden/monitoring/service/UserService.kt
package com.jayden.monitoring.service

import com.jayden.monitoring.domain.User
import com.jayden.monitoring.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserService {

    @Autowired
    lateinit var userRepository: UserRepository

    @Transactional
    fun createUser(name: String): Long {
        val user = User(name = name)
        userRepository.save(user)
        return user?.id!!
    }

    fun getUserList(): List<User> {
        return userRepository.findAll()
    }

    fun getUser(id: Long): User {
        return userRepository.findById(id).orElseThrow { RuntimeException() }
    }
}
